### PR TITLE
Revert "Skip StatefulSet e2e tests if this resource is not found"

### PR DESCRIPTION
**Pull request by foxish**
Replaces https://github.com/kubernetes/kubernetes/pull/37398

### CODE REVIEW

File: test/e2e/petset.go [hunk 1]
```diff
@@ -408,7 +403,7 @@ var _ = framework.KubeDescribe("Stateful Set recreate", func() {
 	petPodName := "web-0"
 
 	BeforeEach(func() {
-		framework.SkipIfMissingResource(f.ClientPool, StatefulSetGroupVersionResource, f.Namespace.Name)
+		framework.SkipUnlessProviderIs("gce", "vagrant")
```
Comment (foxish) [inline test/e2e/petset.go]: Add in GKE here? @janetkuo

Comment (foxish) [inline test/e2e/petset.go]: Looks like they're running in GKE, under https://k8s-testgrid.appspot.com/google-gke#gci-gke. So, we should continue to have it enabled.

**Review by janetkuo — APPROVED** [review]

**Comment by foxish** [comment]
Still testing the old commit for some reason. 
@k8s-bot test this #IGNORE

**Pull request merged**